Extension/src/LanguageServer/extension.ts

```diff
@@ -1062,6 +1110,37 @@ function handleMacCrashFileRead(err: NodeJS.ErrnoException | undefined | null, d
     logMacCrashTelemetry(data);
 }
 
+async function handleCrashFileRead(crashDirectory: string, crashFile: string, err: NodeJS.ErrnoException | undefined | null, data: string): Promise<void> {
+    if (err) {
+        if (err.code === "ENOENT") {
+            return; // ignore known issue
+        }
+        return logCppCrashTelemetry("readFile: " + err.code);
+    }
+
+    const lines: string[] = data.split("\n");
+    data = crashFile + "\n";
+    const filtPath: string | null = which.sync("c++filt", { nothrow: true });
+    for (let line of lines)
+    {
+        line = line.trim();
+        if (filtPath && line.startsWith("_")) {
+            line = (await util.spawnChildProcess(filtPath, [line])).output;
+            line = line.replace(/std::(?:__1|__cxx11|basic_)/g, "std::"); // simplify std stuff.
+        }
+        data += line + "\n";
+    }
+
+    if (data.length > 8192) { // The API has an 8k limit.
+        data = data.substring(0, 8189) + "...";
+    }
+
+    logCppCrashTelemetry(data);
+
+    await util.deleteFile(path.resolve(crashDirectory, crashFile));
+    void util.deleteDirectory(crashDirectory);
```

Comment:
I'm not sure if we need to delete the directory if it's in the `storagePath`. But if we do, you need to `.catch()` on this in case it rejects the promise.

Reply:
It's not in the storage path, it's in a tmp path